io-sim-classes/src/Control/Monad/Class/MonadSTM/Strict.hs

```diff
@@ -79,6 +82,9 @@ data StrictTVar m a = StrictTVar
    , tvar      :: !(LazyTVar m a)
    }
 
+instance Eq (LazyTVar m a) => Eq (StrictTVar m a) where
+    (==) = on (==) tvar
```

Comment:
Why do we need the superclass constraint?

We already know that `TVar s a` has an `Eq` instance, and that instance has no further preconditions. So we should be able to do this directly without undecidable instances or this superclass constraint.

What am I missing?

Reply:
> We already know that TVar s a has an Eq instance
We don't.  I tried to add `forall a. Eq (TVar_ stm a)` as a constraint to `MonadSTMTx` but it did not work, is there some other trick to add a constraint on an associated type family?